### DISCUSSION TITLE
Per game hotkeys?

### DISCUSSION THREAD

**Discussion by danielristic** [eduardofilo/RG350_adam_image#112] [RetroArch topics]
While the default hotkeys are great, they sometimes prevent from using in-game saves (eg. with Link's Awakening on GB and GBC, a very popular Zelda game that requires to press `Start+Select+A+B` at the same time to show the save menu).

As pressing `Start+Select` closes the game, I thought it would be as simple as activating retroarch's `Confirm Quit` and then I'd be able to use the in-game save.

As expected, it no longer closed the game but I then noticed that `Select+A` and `Select+B` were also mapped to pause and reset emulation, so I tried to remove them only for that specific game using `Quick Menu > Overrides > Save Game Override` but that didn't work. As the `Hotkey` settings are in the `Input` section I then thought that maybe it's a control override so I tried `Quick Menu > Controls > Save game remap file` but that didn't work either.

The only was I managed to make this work is by launching retroarch from the Apps and Games section and then save the global config from `Config File > Save Current Configuration` but then all games are affected.

**TLDR; Is it possible to change hotkeys only for a specific game?**

**Comment by eduardofilo** [comment]
I have tried to reproduce the situation to find a solution, but I do not know the game and therefore I do not know if the combination `Select + Start + A + B` should work from the beginning (nothing appears to me after having deactivated the hotkeys that interfere) or from a certain point in the game.

**Comment by danielristic** [comment]
The in-game save is available almost immediately, from the moment you're able to show the pause menu when pressing start, which happens after you've exited the house (you need to talk to the man so he gives you the shield first).

This pause menu also suffers a separate but related issue: when holding select in the pause menu you should be able to see a submenu with extra information but due to the conflicting hotkey the submenu shows and disappears almost immediately. I've been able to mitigate this issue by setting the `hotkey enable delay` setting to a much larger value (120).

**Comment by eduardofilo** [comment]
OK. I have managed to access that menu. Certainly, it should be enough to override at the game level, but it seems that there is some bug (or feature that we do not know about) that causes deactivations of the hotkeys of the A and B keys (Pause and Reset) to not be saved in the override. I have managed to add them manually in the attached override.

```
input_pause_toggle_btn = "nul"
input_reset_btn = "nul"
```

You will have to rename the file to match your ROM, and place it in the path `/media/data/local/home/.retroarch/config/Gambatte`

Anyway, although it works for me, it is difficult to access that menu. I don't know if the solution will seem functional to you.

[Legend of Zelda, The - Link's Awakening DX (U-ESP) (V1.2).cfg.zip](https://github.com/eduardofilo/RG350_adam_image/files/7656551/Legend.of.Zelda.The.-.Link.s.Awakening.DX.U-ESP.V1.2.cfg.zip)
